File: YangTuanAnh/lesson_1/Linked List Cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        if (head==NULL) return false;
        if (head->next==NULL) return false;
        ListNode* fast = head, *slow = head;
        while (slow && fast) {
            slow = slow->next;
            fast = fast->next;
            if (fast) fast = fast->next;
            if (slow==fast) return true;
        }
        return false;
    }
};

File: Homework/Le_Trong/src/lesson_01/linked_list_cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let slow = head;
    let quick = head;

    while (quick != null && quick.next != null) {
        slow = slow.next;
        quick = quick.next.next;

        if (slow === quick) {
            return true
        }
    }

    return false;
};
